software/contrib/turing_machine.py

```diff
@@ -0,0 +1,254 @@
+"""
+ A script meant to recreate the Music Thing Modular Turning Machine Random Sequencer as faithfully as possible on the 
+ EuroPi hardware using bit shift operations to mimic the analog shift register.
+
+din - clock
+ain - cv control over the big knob, added to the knobs value
+k1 - the big knob (probability that the sequence changes)
+k2 - output scale (0-10v)  or sequence length (2-16 steps)
+b1 - write (clear bits)
+b2 - change k2 function
+cv1 - pulse 1
+cv2 - pulse 2
+cv3 - pulse 4
+cv4 - pulse 1 & 2
+cv5 - pulse 2 & 4
+cv6 - sequence out
+
+If you'd like to use different bits for the pulse outputs you can update the ``CVX_PULSE_BIT`` constants below.
```

Comment:
Since scripts don't get their docs rendered by Sphinx, I think it would be better to use Markdown syntax, which I believe VS Code would render via docstring static analysis (untested, I could be wrong).